components/brave_shields/browser/domain_block_navigation_throttle.cc

```diff
@@ -61,27 +60,28 @@ std::unique_ptr<DomainBlockNavigationThrottle>
 DomainBlockNavigationThrottle::MaybeCreateThrottleFor(
     content::NavigationHandle* navigation_handle,
     AdBlockService* ad_block_service,
-    AdBlockCustomFiltersService* ad_block_custom_filters_service,
+    AdBlockCustomFiltersSourceProvider* ad_block_custom_filters_source_provider,
```

Comment:
this is definitely not right, why are we passing in the source provider?

Reply:
It's the same reason as https://github.com/brave/brave-core/pull/10994#discussion_r759792047.

Reply:
I think you're missing what I'm saying, the SourceProvider shouldn't be passed to other classes, they should only deal with the service.

Reply:
I guess my reasoning is that there isn't a `CustomFiltersService` in the current implementation; just the engine and SourceProvider. The method to add the new custom filter could be added to AdBlockService in the current implementation, or adding CustomFiltersService to handle it could work too.